deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -517,7 +516,14 @@ def is_moe_group(self, group):
         return 'moe' in group and group['moe']
 
     def _configure_moe_settings(self):
-        assert self.contiguous_gradients, "Contiguous Gradients in ZeRO Stage 2 must be set to True for MoE. Other code paths are not tested with MoE"
+        # if we're using ZeRO stage 2, ensure contiguous gradients are used
```

Comment:
This difference seems unrelated to communication backend. Please double check.

Reply:
This arose from my Megatron-DeepSpeed testing. This assertion was failing MoE with ZeRO-1, which is a supported mode. I discussed this with @awan-10 and we agreed on this fix.

Would you like me to create a separate PR for this or can it be bundled with this one as-is?

Reply:
Yes, please. A separate PR will help track things in future.

Reply:
I reverted this commit and created a separate PR here: https://github.com/microsoft/DeepSpeed/pull/2007